pilot/cmd/pilot-discovery/main.go

```diff
@@ -103,6 +103,16 @@ func init() {
 		"Controller resync interval")
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.ControllerOptions.DomainSuffix, "domain", "cluster.local",
 		"DNS domain suffix")
+
+	defaultID := "<unknown>"
+	podName := os.Getenv("POD_NAME")
+	podNamespace := os.Getenv("POD_NAMESPACE")
+	if podName != "" && podNamespace != "" {
+		defaultID = fmt.Sprintf("kubernetes://%s.%s", podName, podNamespace)
```

Comment:
Not sure this is the best way to infer where pilot gets its config - you can have a pilot running in k8s but using a different store. You can use the registries (maybe) - but since configs can be loaded from different places.

What is the use of this id ?

Reply:
The instance ID is the identifying information of the pilot instance that is _consuming_ k8s CRD resources. It doesn't indicate the source of config, though that is implied because Pilot only consumes _CRDS_ from one cluster AFIACT.